### DISCUSSION TITLE
Device status updates

### DISCUSSION THREAD

**Discussion by s0129** [tsightler/ring-mqtt#578] [Support Q&A] [answered]
Is it intended for the status topic to be updated regularly or does it only get updated on changes?
e.g. ring/<location_id>/<product_category>/<device_id>/status

I am finding it gets updated on a regular basis for the first few minutes and then doesn't get updated until something goes offline.
As the status isn't sent with retain I am finding it hard for a client to tell if devices are online/offline unless they were listening from when the ring-mqtt was started.

I am doing something wrong or is there is a way to request a status update?

**Comment by tsightler** [comment, marked as answer]
There are no status updates unless state changes.  I've considered using retain only for status messages but, honestly, it doesn't help much since you still wouldn't know the current state and using retain for state messages is a mess.  Retain just caused so many problems with users in the early days of this project that I just swore it off completely.

During startup all state updates, including online/offline state, are sent every 30 seconds for 3 minutes.  This is mostly to work around the fact that HA sometimes sends its online message before it's really ready to accept device discovery and state update information.

There's no direct way to request a state update, but the code does listen for online/offline events from Home Assistant on the homeassistent/status topic, and uses this to know when it needs to resend all state updates, so you can do the same thing with your code.  Note that any sending of "online" will cause the 30 seconds updates to occure for 3 minutes.

**Reply by s0129** [reply]
Thanks that is useful to know. I can use the homeassistant status topic to trigger state updates and I think this will work nicely for what I am trying to do.
Thanks!
